### ISSUE TITLE
Deleting a user doesn't remove all associations (such as payment modules)

### ISSUE THREAD

**Issue by mjgardes** [elplatt/seltzer#344]
A couple months ago, a user was added to the i3Detroit CRM with the incorrect email address.  That account was deleted and recreated with the correct information, however the delete operation left the database polluted with references to the same name and the old CID.  It appears that the CID was only deleted from one table and still showed up in the reports and caused a great deal of extra work for the treasurer as he had to manually enter every payment for this user since then.

**Comment by chris18890** [comment]
so the dev branch then?

**Comment by mjgardes** [comment]
Looks like we last pulled from master.

**Comment by chris18890** [comment]
ok, cause on dev when a contact is deleted, everything but the stuff in the payment table is removed.

Also, surely you could've just changed the email address?

**Comment by mjgardes** [comment]
Of course.  If I had been the one doing member administration at the time,
I probably would have.

**Comment by chris18890** [comment]
well, being on the latest dev release might solve some of the issues.

However, removing payments when deleting a user would require a fair bit of new code in the payment module.

There is also the question, for audit/bookkeeping purposes, should you be deleting payment info?

thoughts @elplatt @mattoehrlein @andrewbolster @dylan1312 ?

**Comment by mjgardes** [comment]
I wouldn't want to remove any payments from the database.  Missing payments
are a nightmare to track down.  The important thing is to make sure that
payments can be associated with the correct CID not sent to a deleted user,
and I don't think editing the database manually is the right way to do that.

**Comment by chris18890** [comment]
from what I can see, that is the case - the payments remain in the database, & the CID is still saved in the "credit" & "debit" columns, but because the member has been deleted it no longer resolves. Personally, I'd say that in order to preserve records member should not be deleted but rather just set to have no active membership
